go/vumitools/conversation/utils.py

```diff
@@ -530,21 +422,29 @@ def get_contact_keys(self):
     @Manager.calls_manager
     def get_inbound_throughput(self, sample_time=300):
         """
-        Calculate how many inbound messages per minute we've been
-        doing on average.
+        Calculate how many inbound messages per minute we've been doing on
+        average.
         """
-        count = yield self.mdb.cache.count_inbound_throughput(
-            self.batch.key, sample_time)
+        inbounds = yield self.mdb.cache.get_inbound_message_keys(
+            self.batch.key, with_timestamp=True)
```

Comment:
Will this iterate through all keys in the batch? Perhaps we should limit keys to the last ``sample_time`` if it does?